magda-web-client/src/Components/Dataset/Add/DatasetAddFilesPage.tsx

```diff
@@ -49,10 +49,29 @@ class DatasetAddFilesPage extends React.Component<
         });
     }
 
-    addFiles = async (fileList: FileList) => {
+    private isDirItem(idx: number, event: any) {
+        try {
+            if (!event) return false;
+            // --- access web browser API to test whether it's a directory
+            const rawEntry = event.dataTransfer.items[idx].webkitGetAsEntry();
```

Comment:
should this be `getAsEntry() || webkitGetAsEntry()` for future proofing? https://developer.mozilla.org/en-US/docs/Web/API/DataTransferItem/webkitGetAsEntry

Reply:
👍 Updated accordingly